Code/Framework/AzCore/AzCore/Math/Sfmt.cpp

```diff
@@ -538,24 +535,16 @@ namespace AZ
     //=========================================================================
     AZ::u64 Sfmt::Rand64()
     {
-        int index = m_index.fetch_add(2);
-        if (index >= (SfmtInternal::N32 - 1))
+        AZStd::lock_guard<decltype(m_generationMutex)> lock(m_generationMutex);
```

Comment:
repeat: Class Template argument deduction can be used here.
```suggestion
        AZStd::scoped_lock lock(m_generationMutex);
```